compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/WindowState.desktop.kt

```diff
@@ -227,20 +229,25 @@ private class WindowStateImpl(
                     it.position.isSpecified,
                     it.position.x.value,
                     it.position.y.value,
-                    it.size.width.value,
-                    it.size.height.value,
+                    it.size.isSpecified,
+                    it.size.takeOrElse { DpSize.Zero }.width.value,
+                    it.size.takeOrElse { DpSize.Zero }.height.value,
                 )
             },
             restore = { state ->
                 WindowStateImpl(
-                    placement = WindowPlacement.values()[state[0] as Int],
+                    placement = WindowPlacement.entries[state[0] as Int],
                     isMinimized = state[1] as Boolean,
                     position = if (state[2] as Boolean) {
                         WindowPosition((state[3] as Float).dp, (state[4] as Float).dp)
                     } else {
                         unspecifiedPosition
                     },
-                    size = DpSize((state[5] as Float).dp, (state[6] as Float).dp),
+                    size = if (state[5] as Boolean) {
```

Comment:
Let's extend WindowStateTest.`save state` test or write a new one for the bug case